### PR TITLE
Add count Vec<u8> Specializations

### PR DESCRIPTION
**Pull request by wcampbell0x2a**
* For Vec<u8> when using count, specialize into reading the bytes all
at once

See #462

### CODE REVIEW

File: src/attributes.rs [hunk 1]
```diff
@@ -47,6 +47,7 @@ enum DekuEnum {
 | [bytes_read](#bytes_read) | field | Set the field representing the number of bytes to read into a container
 | [until](#until) | field | Set a predicate returning when to stop reading elements into a container
 | [read_all](#read_all) | field | Read until [reader.end()] returns `true`
+| [read_exact](#read_exact) | field | Read exactly this many bytes into a `Vec<u8>`
```
Comment (sharksforarms) [inline src/attributes.rs:50]: maybe we could have a separate table/section for specializations

File: deku-derive/src/lib.rs [hunk 2]
```diff
@@ -431,6 +432,9 @@ struct FieldData {
     /// tokens providing the number of bytes for the length of the container
     bytes_read: Option<TokenStream>,
 
+    /// read exactly that many bytes
+    read_exact: Option<TokenStream>,
```
Comment (sharksforarms) [inline deku-derive/src/lib.rs:436]: thinking about this now, could we not specialize `count` for `Vec<u8>` ?

Comment (wcampbell0x2a) [reply deku-derive/src/lib.rs]: Yah, proc-macro specializations aren't my favorite, but we don't really have other options... Updated all the code

**Review by sharksforarms — DISMISSED** [review]
Nice!

**Review by sharksforarms — APPROVED** [review]
Thanks!

**Pull request merged**